scripts/types/fixTypes.ts

```diff
@@ -1,32 +1,91 @@
 /**
- * -------------------------------------------------------
+ * -----------------------------------------------------------------
+ * The file is used to fix various issues with the types
+ * - Add reference paths to the mixins
+ * - Replace the dist triple slash reference with the webgpu types
+ * - Copy over the Shaders.d.ts file to the lib folder for
+ *   proper frag,vert,wgsl support
+ * -----------------------------------------------------------------
+ */
+
+import fs from 'fs';
+import glob from 'glob';
+import path from 'path';
+
+/**
  * Get every d.ts file in the src folder and add a
  * reference path to it in the lib/index.d.ts file
  *
  * This allows us to define our PixiMixins in the src folder
- *
  * e.g. /// <reference path="./app/ApplicationMixins.d.ts" />
- * -------------------------------------------------------
  */
+function addMixinReferencePaths()
+{
+    const globPath = `${path.join(process.cwd(), './src')}/**/*.d.ts`;
+    const files = glob.sync(globPath);
 
-import fs from 'fs';
-import glob from 'glob';
-import path from 'path';
+    const lines: string[] = [];
 
-const globPath = `${path.join(process.cwd(), './src')}/**/*.d.ts`;
-const files = glob.sync(globPath);
+    files.forEach((file) =>
+    {
+        const name = file.split(`${process.cwd()}/src`)[1];
 
-const lines: string[] = [];
+        lines.push(`/// <reference path=".${name}" />`);
+    });
 
-files.forEach((file) =>
+    const filePath = path.join(process.cwd(), './lib/index.d.ts');
+    const contents = fs.readFileSync(filePath, 'utf8');
+    const updatedContents = `${lines.join('\n')}\n${contents}`;
+
+    fs.writeFileSync(filePath, updatedContents);
+}
+
+/**
+ * For whatever reason when we build the types we get /// <reference types="dist" />
```

Comment:
Would be nice to know why this is happening. TS bug or are we implementing something incorrectly?

Reply:
agree! do we have any idea?

Reply:
I currently have no idea what is causing this, but will add a ticket to look into it some more